src/Credentials/CredentialProvider.php

```diff
@@ -278,6 +286,64 @@ public static function assumeRole(array $config=[])
         return new AssumeRoleCredentialProvider($config);
     }
 
+    /**
+     * Credential provider that creates credentials by assuming role from a
+     * Web Identity Token
+     *
+     * @param array $config Array of configuration data
+     * @return callable
+     * @see Aws\Credentials\AssumeRoleWithWebIdentityCredentialProvider for
+     * $config details.
+     */
+    public static function assumeRoleWithWebIdentityCredentialProvider(array $config = [])
+    {
+        return function () use ($config) {
+            $arnFromEnv = getenv(self::ENV_ARN);
+            $tokenFromEnv = getenv(self::ENV_TOKEN_FILE);
+            $stsClient = isset($config['stsClient'])
+                ? $config['stsClient']
+                : null;
+
+            if ($tokenFromEnv && $arnFromEnv) {
+                $sessionName = getenv(self::ENV_ROLE_SESSION_NAME)
+                    ? getenv(self::ENV_ROLE_SESSION_NAME)
+                    : null;
+                $provider = new AssumeRoleWithWebIdentityCredentialProvider([
+                    'RoleArn' => $arnFromEnv,
+                    'WebIdentityTokenFile' => $tokenFromEnv,
+                    'SessionName' => $sessionName,
+                    'client' => $stsClient
+                ]);
+
+                return $provider();
+            }
+
+            $profileName = getenv(self::ENV_PROFILE) ?: 'default';
+            $filename = isset($config['filename'])
+                ? $config['filename']
+                : self::getHomeDir() . '/.aws/credentials';
```

Comment:
This doesn't seem specifically addressed by the SEP, but should `/.aws/config` also be looked at?